### ISSUE TITLE
Spurious W3301: min(value, min(iterable)) does not do the same thing as min(value, iterable)

### ISSUE THREAD

**Issue by zackw** [PyCQA/pylint#9923]
### Bug description

`min(iterable)` produces the minimum element of the iterable. `min(value, value, ...)` produces the minimum of all its arguments; it does *not* notice iterables among the values. Therefore `min(value, min(iterable))` is a useful thing to do, but `min(value, iterable)` will typically throw a TypeError because `int` and `list[int]` (for instance) are not comparable. The same applies to `max()`.

To sum up: The code below is the natural way to take the minimum/maximum of an iterable plus some other values, and it should not trigger any W3301 warnings.

```python
#pylint:disable=missing-docstring
#pylint:enable=nested-min-max

def bounds(lo, hi, elements):
    return (
        min(lo, hi, min(elements)),
        max(lo, hi, max(elements)),
    )
```

### Command used

```shell
pylint a.py
```

### Pylint output

```none
************* Module a
a.py:18:8: W3301: Do not use nested call of 'min'; it's possible to do 'min(lo, hi, elements)' instead (nested-min-max)
a.py:19:8: W3301: Do not use nested call of 'max'; it's possible to do 'max(lo, hi, elements)' instead (nested-min-max)
```

### Expected behavior

Do not issue W3301 when the inner call(s) to min/max pass only one argument.

### Pylint version

```none
pylint 3.2.7
astroid 3.2.4
Python 3.8.19 (default, Sep  9 2024, 11:01:37)
[GCC 13.3.1 20240614]
```

**Comment by zenlyj** [comment]
Thank you for this, i was able to reproduce the issue. `min(value, min(iterable))` is pretty useful, an alternative without any nested calls is `min(value, *iterable)`.

After looking into the implementation, there is already a line to check whether the outer call has one argument. Perhaps the condition can be extended to include the suggestion, to check whether the inner call has one argument as well.

https://github.com/pylint-dev/pylint/blob/0687c851e14cf187bb213128c4f23feecd0105ac/pylint/checkers/nested_min_max.py#L60-L72

**Comment by RichardJBlack** [comment]
This hit me.  Do not complain about min(int, min(list[int])) or else do not suggest code that will fail with `TypeError: '<' not supported between instances of 'list' and 'int'`.

**Comment by zackw** [comment]
> an alternative to min(value, min(iterable)) is min(value, *iterable), without the nested call

In the situation that prompted the original bug report, the iterable was a generator that could yield millions of elements; stuffing them all into a call tuple would be undesirable.

> not rais[ing] any message when the inner call has only one argument could lead to false negatives like [example]

hm.  The diagnostic _is_ pointing at a real footgun in the case of your example...

```
>>> min(1)
Traceback (most recent call last):
  File "<python-input-0>", line 1, in <module>
    min(1)
    ~~~^^^
TypeError: 'int' object is not iterable
```

Tempted to file a bug on CPython over that, tbh; the mathematical Right Thing would be to return the only object supplied as an argument, when it isn't iterable but _is_ comparable.

However, _despite that_, for untyped code I would much rather have this false negative than the current false positive.  If you're going to make the diagnostic more precise when type information is available, turn it back _on_ when the single argument to the inner min/max call is known _not_ to be iterable.

**Comment by Pierre-Sassoulas** [comment]
I think we need a better suggestion if we detect an iterable as pointed out by @zenlyj (thank you for triaging :) ).
